x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -302,7 +302,8 @@ static String getPutMappingIndexOrAlias(PutMappingRequest request, Set<String> a
     static boolean allowsRemoteIndices(IndicesRequest request) {
         return request instanceof SearchRequest || request instanceof FieldCapabilitiesRequest
                 || request instanceof GraphExploreRequest || request instanceof ResolveIndexAction.Request
-                || request instanceof OpenPointInTimeRequest;
+                || request instanceof OpenPointInTimeRequest
+                || request.getClass().getCanonicalName().equals("org.elasticsearch.xpack.eql.action.EqlSearchRequest");
```

Comment:
This needs to be double checked with the security team.

Reply:
Will do.

Reply:
@albertzaharovits the integration test has been extended to be done with the security enabled, let me know if anything else is missing from your side. Thanks.

Reply:
LGTM to me, thank you @bpintea .